### PR TITLE
fix: only include needed imported bindings in golang

### PR DESCRIPTION
**Pull request by Niraj-Kamdar**

### CODE REVIEW

File: packages/schema/bind/src/bindings/golang/wasm/templates/env-type/Env%type%Serialization-go.mustache [hunk 1]
```diff
@@ -12,8 +12,8 @@ func serialize{{#toUpper}}{{type}}{{/toUpper}}(value *{{#toUpper}}{{type}}{{/toU
 func write{{#toUpper}}{{type}}{{/toUpper}}(writer msgpack.Write, value *{{#toUpper}}{{type}}{{/toUpper}}) {
 	writer.WriteMapLength({{properties.length}})
 	{{#properties}}
-	writer.Context().Push("{{#toUpper}}{{#handleKeywords}}{{name}}{{/handleKeywords}}{{/toUpper}}", "{{#toWasm}}{{toGraphQLType}}{{/toWasm}}", "writing property")
-	writer.WriteString("{{#toUpper}}{{#handleKeywords}}{{name}}{{/handleKeywords}}{{/toUpper}}")
+	writer.Context().Push("{{#toLower}}{{#handleKeywords}}{{name}}{{/handleKeywords}}{{/toLower}}", "{{#toWasm}}{{toGraphQLType}}{{/toWasm}}", "writing property")
+	writer.WriteString("{{#toLower}}{{#handleKeywords}}{{name}}{{/handleKeywords}}{{/toLower}}")
```
Comment (dOrgJelli) [inline packages/schema/bind/src/bindings/golang/wasm/templates/env-type/Env%type%Serialization-go.mustache]: This is incorrect, the name should be 1:1 with what's in the schema, so just `{{name}}`.

File: packages/schema/bind/src/bindings/golang/wasm/templates/imported/env-type/Env%type%Serialization-go.mustache [hunk 2]
```diff
@@ -15,8 +15,8 @@ func serialize{{#toUpper}}{{type}}{{/toUpper}}(value *{{#toUpper}}{{type}}{{/toU
 func write{{#toUpper}}{{type}}{{/toUpper}}(writer msgpack.Write, value *{{#toUpper}}{{type}}{{/toUpper}}) {
 	writer.WriteMapLength({{properties.length}})
 	{{#properties}}
-	writer.Context().Push("{{#toUpper}}{{#handleKeywords}}{{name}}{{/handleKeywords}}{{/toUpper}}", "{{#toWasm}}{{toGraphQLType}}{{/toWasm}}", "writing property")
-	writer.WriteString("{{#toUpper}}{{#handleKeywords}}{{name}}{{/handleKeywords}}{{/toUpper}}")
+	writer.Context().Push("{{#toLower}}{{#handleKeywords}}{{name}}{{/handleKeywords}}{{/toLower}}", "{{#toWasm}}{{toGraphQLType}}{{/toWasm}}", "writing property")
+	writer.WriteString("{{#toLower}}{{#handleKeywords}}{{name}}{{/handleKeywords}}{{/toLower}}")
```
Comment (dOrgJelli) [inline packages/schema/bind/src/bindings/golang/wasm/templates/imported/env-type/Env%type%Serialization-go.mustache]: This is incorrect, the name should be 1:1 with what's in the schema, so just {{name}}.

File: packages/schema/bind/src/bindings/golang/wasm/templates/imported/object-type/Object%type%Serialization-go.mustache [hunk 3]
```diff
@@ -15,8 +15,8 @@ func serialize{{#toUpper}}{{type}}{{/toUpper}}(value *{{#toUpper}}{{type}}{{/toU
 func write{{#toUpper}}{{type}}{{/toUpper}}(writer msgpack.Write, value *{{#toUpper}}{{type}}{{/toUpper}}) {
 	writer.WriteMapLength({{properties.length}})
 	{{#properties}}
-	writer.Context().Push("{{#toUpper}}{{#handleKeywords}}{{name}}{{/handleKeywords}}{{/toUpper}}", "{{#toWasm}}{{toGraphQLType}}{{/toWasm}}", "writing property")
-	writer.WriteString("{{#toUpper}}{{#handleKeywords}}{{name}}{{/handleKeywords}}{{/toUpper}}")
+	writer.Context().Push("{{#toLower}}{{#handleKeywords}}{{name}}{{/handleKeywords}}{{/toLower}}", "{{#toWasm}}{{toGraphQLType}}{{/toWasm}}", "writing property")
+	writer.WriteString("{{#toLower}}{{#handleKeywords}}{{name}}{{/handleKeywords}}{{/toLower}}")
```
Comment (dOrgJelli) [inline packages/schema/bind/src/bindings/golang/wasm/templates/imported/object-type/Object%type%Serialization-go.mustache]: This is incorrect, the name should be 1:1 with what's in the schema, so just {{name}}.

File: packages/schema/bind/src/bindings/golang/wasm/templates/object-type/Object%type%Serialization-go.mustache [hunk 4]
```diff
@@ -15,8 +15,8 @@ func serialize{{#toUpper}}{{type}}{{/toUpper}}(value *{{#toUpper}}{{type}}{{/toU
 func write{{#toUpper}}{{type}}{{/toUpper}}(writer msgpack.Write, value *{{#toUpper}}{{type}}{{/toUpper}}) {
 	writer.WriteMapLength({{properties.length}})
 	{{#properties}}
-	writer.Context().Push("{{#toUpper}}{{#handleKeywords}}{{name}}{{/handleKeywords}}{{/toUpper}}", "{{#toWasm}}{{toGraphQLType}}{{/toWasm}}", "writing property")
-	writer.WriteString("{{#toUpper}}{{#handleKeywords}}{{name}}{{/handleKeywords}}{{/toUpper}}")
+	writer.Context().Push("{{#toLower}}{{#handleKeywords}}{{name}}{{/handleKeywords}}{{/toLower}}", "{{#toWasm}}{{toGraphQLType}}{{/toWasm}}", "writing property")
+	writer.WriteString("{{#toLower}}{{#handleKeywords}}{{name}}{{/handleKeywords}}{{/toLower}}")
```
Comment (dOrgJelli) [inline packages/schema/bind/src/bindings/golang/wasm/templates/object-type/Object%type%Serialization-go.mustache]: This is incorrect, the name should be 1:1 with what's in the schema, so just {{name}}.

File: packages/templates/wasm/golang/build.py [hunk 5]
```diff
@@ -0,0 +1,106 @@
+import os
+import subprocess
+import re
+
+ACCEPTED_IMPORTS = {'wrap', 'env'}
+
+def build_wam_module():
```
Comment (dOrgJelli) [inline packages/templates/wasm/golang/build.py]: this file shouldn't exist, all of this should be done through the CLI's build image

File: packages/templates/wasm/golang/go.mod [hunk 6]
```diff
@@ -2,6 +2,6 @@ module example.com/template-wasm-go
 
 go 1.18
 
-require github.com/polywrap/go-wrap wrap-0.1
+require github.com/polywrap/go-wrap v0.0.0-20230712212127-6895977d63c2
```
Comment (dOrgJelli) [inline packages/templates/wasm/golang/go.mod]: please leave as `wrap-0.1`

File: packages/templates/wasm/golang/wasm-target.json [hunk 7]
```diff
@@ -0,0 +1,27 @@
+{
```
Comment (dOrgJelli) [inline packages/templates/wasm/golang/wasm-target.json]: why is this in the template project?

**Review by dOrgJelli — CHANGES_REQUESTED** [review]

File: packages/templates/wasm/golang/go.mod (hunk 6 above)
Comment (Niraj-Kamdar) [reply packages/templates/wasm/golang/go.mod]: It automatically changes to the commit version when we run `go mod tidy` but I will update it.

Comment (Niraj-Kamdar) [reply packages/templates/wasm/golang/go.mod]: I think We should use tag for the releases

**Review by dOrgJelli — APPROVED** [review]

File: packages/templates/wasm/golang/go.mod (hunk 6 above)
Comment (dOrgJelli) [reply packages/templates/wasm/golang/go.mod]: agreed

**Pull request merged**